src/Features/Core/Portable/ChangeSignature/ChangeSignatureTelemetryLogger.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Internal.Log;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class ChangeSignatureLogger
+    {
+        private const string Maximum = nameof(Maximum);
+        private const string Minimum = nameof(Minimum);
+        private const string Mean = nameof(Mean);
+
+        private static readonly LogAggregator s_logAggregator = new LogAggregator();
+        private static readonly StatisticLogAggregator s_statisticLogAggregator = new StatisticLogAggregator();
+        private static readonly HistogramLogAggregator s_histogramLogAggregator = new HistogramLogAggregator(bucketSize: 50, maxBucketValue: 1000);
```

Comment:
I'm not sure about these values.

Reply:
bucketSize is the range of one bucket, starts from 0, i.e. [0,50), [50, 100), etc., maxBucketValue is the largest number it tracks, i.e. any value larger than 1000 will be counted in a single bucket of range [1000,)

---
In reply to: [411518337](https://github.com/dotnet/roslyn/pull/43504#discussion_r411518337) [](ancestors = 411518337)

Reply:
I think the value you used are identical to what I used for completion, you can see an example here (at the bottom)
[https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_dashboards/dashboard/73a410a4-f68c-404f-938a-ab5f821532f9](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_dashboards/dashboard/73a410a4-f68c-404f-938a-ab5f821532f9)

---
In reply to: [412431112](https://github.com/dotnet/roslyn/pull/43504#discussion_r412431112) [](ancestors = 412431112,411518337)

Reply:
Thanks. I'm going to change to 1-second buckets up to 30 seconds.